### DISCUSSION TITLE
drive two different SSD1322

### DISCUSSION THREAD

**Discussion by nibble-bit-dev** [olikraus/u8g2#1431] [Q&A]
Hi,
I'm using u8g2 library with an SSD1322 dispaly.
All works perfectly.

No I'd want to add a second SSD1322  using the same set of pins, except clearly for the /CS pin.
I usually do that by spi protocol.

Now, I tried by the code as below way but without success:

My first attempt was define two different 'u8x8_avr_gpio_and_delay' function, 
where I defined two different DISPLAY_CS_PIN definition.

DISPLAY_CS_PIN_0 //1st SSD1322
DISPLAY_CS_PIN_1 // 2nd SSD1322

//the code below is ONLY FOR TEST PURPOSE ...please do not consider as good programming..

uint8_t u8x8_avr_gpio_and_delay_0(u8x8_t *u8x8, uint8_t msg, uint8_t arg_int, void *arg_ptr);
uint8_t u8x8_avr_gpio_and_delay_1(u8x8_t *u8x8, uint8_t msg, uint8_t arg_int, void *arg_ptr);

also I defined  two different  u8g2_t data type.

u8g2_t u8g2_0;
u8g2_t u8g2_1;

Then I created two init functions: 

void ssd1322_init_0()
{
	
	u8g2_Setup_ssd1322_nhd_256x64_f(&u8g2_0, U8G2_R0, u8x8_byte_4wire_sw_spi, u8x8_avr_gpio_and_delay_0 );
	u8g2_InitDisplay(&u8g2_0);
	u8g2_SetPowerSave(&u8g2_0, 0);	
}

void ssd1322_init_1()
{
	
	u8g2_Setup_ssd1322_nhd_256x64_f(&u8g2_1, U8G2_R0, u8x8_byte_4wire_sw_spi, u8x8_avr_gpio_and_delay_1 );	
	u8g2_InitDisplay(&u8g2_1);
	u8g2_SetPowerSave(&u8g2_1, 0);	
}

This code does not work. And my write string function does not work.

The first parameter can be u8g2_0 or u8g2_1:

void ssd1322_write_string(u8g2_t *u8g2, char* str)
{
	int ln = strlen(str);
	u8g2_ClearBuffer(&u8g2);
	u8g2_SetFont(&u8g2, u8g2_font_ncenB14_tr);
	u8g2_DrawStr(&u8g2, 1, ln, str);
	u8g2_SendBuffer(&u8g2);
}

Could you suggest me the right way to do this?

Thanks 
nbdev

**Comment by olikraus** [comment]
Do you share the same reset line with both displays? If so you must not drive the reset line with the second display (because it was already reseted with the first init and generating a second reset will erase the init sequence of the first display)

**Reply by nibble-bit-dev** [reply]
Ok I'll try.
But there are some rules in order to drive two display?
I do not know if my pseudo code solution is the right way.

Thanks
nbdev

**Reply by olikraus** [reply]
I have added some code here:
https://github.com/olikraus/u8g2/issues/1351#issuecomment-794597667

Basically you shouldn't interleave operations on both displays, because they share the same memory.
